lib/service/service.go

```diff
@@ -534,13 +554,28 @@ func (process *TeleportProcess) SetExpectedInstanceRole(role types.SystemRole, e
 	process.instanceRoles[role] = eventName
 }
 
+// SetExpectedHostedPluginRole marks a given hosted plugin role as active, storing the name of its associated
+// identity event.
+func (process *TeleportProcess) SetExpectedHostedPluginRole(role types.SystemRole, eventName string) {
```

Comment:
or is it going to be used in followup PRs?

Reply:
Yeah, will be used in a followup ent PR.  Plugins in ent currently register themselves as normal system roles, but that throws the logic here for a loop, so this PR introduces a new category for them to register under so that they can be treated differently.